core/procedures.js

```diff
@@ -306,40 +306,106 @@ Blockly.Procedures.mutateCallers = function(defBlock) {
 
 /**
  * Find the definition block for the named procedure.
- * @param {string} name Name of procedure.
+ * @param {string} procCode The identifier of the procedure to delete.
  * @param {!Blockly.Workspace} workspace The workspace to search.
  * @return {Blockly.Block} The procedure definition block, or null not found.
  */
-Blockly.Procedures.getDefinition = function(name, workspace) {
+Blockly.Procedures.getDefineBlock = function(procCode, workspace) {
```

Comment:
This whole function can be replaced by finding the prototype block and then getting its parent.  Or vice versa.